rllib/core/models/catalog.py

```diff
@@ -224,3 +240,80 @@ def get_tokenizer_config(
                 **{"use_lstm": False, "use_attention": False},
             },
         )
+
+    @classmethod
+    def get_action_dist_cls_dict(
+        cls,
+        action_space: gym.Space,
+        model_config_dict: ModelConfigDict,
+        deterministic: Optional[bool] = False,
+    ) -> Mapping[str, Any]:
+        """Returns a mapping from framework to distribution class.
+
+        You can get the required input dimension for the distribution by calling
+        `action_dict_cls.required_model_output_shape(action_space, model_config_dict)`
+        on the retrieved class. This is useful, because the Catalog needs to find out
+        about the required input dimension for the distribution before the model that
+        outputs these inputs is configured.
+
+        Args:
+            action_space: Action space of the target gym env.
+            model_config_dict: Optional model config.
+            deterministic: Whether to return a Deterministic distribution instead of
+                a diagonal Gaussian one for Box spaces.
+
+        Returns:
+                Mapping from framework to distribution class.
+        """
+        assert not model_config_dict.get("custom_action_dist"), (
+            "Specify a custom action "
+            "distribution by overwriting "
+            "Catalog.get_action_distribution."
+        )
+
+        # Box space -> DiagGaussian OR Deterministic.
+        if isinstance(action_space, Box):
+            if action_space.dtype.name.startswith("int"):
+                # TODO(Artur): Support multi-discrete.
+                raise NotImplementedError(
+                    "MultiDiscrete/integer Box spaces not yet " "supported."
+                )
+            else:
+                if len(action_space.shape) > 1:
+                    raise UnsupportedSpaceException(
+                        "Action space has multiple dimensions "
+                        "{}. ".format(action_space.shape)
+                        + "Consider reshaping this into a single dimension, "
+                        "using a custom action distribution, "
+                        "using a Tuple action space, or the multi-agent API."
+                    )
+                if deterministic:
+                    return {"torch": TorchDeterministic, "tf": TfDeterministic}
```

Comment:
I would predefine all these dicts at the top of this function and then index via framework input arg.